src/services/interactionUtils.ts

```diff
@@ -147,27 +88,91 @@ function newTypeInteraction(
   return undefined
 }
 
-export function createMapInteractions(rennesApp: RennesApp) {
-  const types_interaction = [
-    SelectRoofInteraction,
-    ForbidenClickInteraction,
-    SelectDistrictInteraction,
-  ]
+export function updateMapInteractionsAfterViewChange(rennesApp: RennesApp) {
+  const interactionsStore = useInteractionsStore()
+  const viewStore = useViewsStore()
+  const simulationStore = useSimulationStore()
+
+  // disable ForbiddenClick elsewhere than home
+  if (viewStore.currentView === viewList.home) {
+    interactionsStore.enableInteraction(ForbidenClickInteraction)
+  } else {
+    interactionsStore.disableInteraction(ForbidenClickInteraction)
+  }
+
+  // enable selectRoofInteraction whatever the zoom is on step sunshine
+  if (
+    viewStore.currentView === viewList['step-sunshine'] &&
+    simulationStore.currentStep === 1 &&
+    simulationStore.currentSubStep === 1
+  ) {
+    interactionsStore.enableInteraction(SelectRoofInteraction)
+  }
+
+  // switch between selectRoof and Iris on first pages
+  // roof selection is block by Forbidden on home
+  if (
+    [
+      viewList['home'],
+      viewList['roof-selected-information'],
+      viewList['roof-selection'],
+    ].includes(viewStore.currentView!)
+  ) {
+    manageSelectAndIrisDependingOnZoom(rennesApp)
+  }
+  if (
+    [viewList['roof-selected-information'], viewList['step-sunshine']].includes(
+      viewStore.currentView!
+    )
+  ) {
+    interactionsStore.disableInteraction(SelectDistrictInteraction)
+  }
+}
 
-  types_interaction.forEach((interactionType) => {
-    if (isInteractionToAdd(interactionType)) {
-      if (!isInteractionExist(rennesApp, interactionType)) {
-        rennesApp.maps.eventHandler.featureInteraction.setActive(
-          EventType.CLICK
-        )
-        const interaction = newTypeInteraction(rennesApp, interactionType)
-        if (interaction)
-          rennesApp.maps.eventHandler.addPersistentInteraction(interaction)
-      } else {
-        activeInteraction(rennesApp, interactionType)
+/**
+ * Disable district interaction if too close to the ground
+ * Activate selectRoofInteraction is close enough and selectable to true
+ * @param rennesApp rennesApp
+ */
+export function manageSelectAndIrisDependingOnZoom(rennesApp: RennesApp) {
+  const interactionsStore = useInteractionsStore()
+  const districtStore = useDistrictStore()
+  const mapStore = useMapStore()
+  const DISTANCE_MAX_FOR_SELECTION = 800
+
+  if (mapStore.isInitializeMap) {
+    rennesApp
+      .get3DMap()
+      .getScene()
+      .postRender.addEventListener(() => {
+        const cameraDistance = rennesApp.getCurrentDistance()!
+        if (cameraDistance > DISTANCE_MAX_FOR_SELECTION) {
+          if (interactionsStore.isActive(SelectRoofInteraction)) {
+            interactionsStore.disableInteraction(SelectRoofInteraction)
+          }
+          districtStore.setCanBeDisplayed(true)
```

Comment:
Can you put the line "districtStore.setCanBeDisplayed(true)" inside the previous if please
postRender runs continuously and districtStore.setCanBeDisplayed(true) is call each time so.